### DISCUSSION TITLE
【持续更新】微信群 Q&A 汇总

### DISCUSSION THREAD

**Discussion by antonia0912** [mlc-ai/mlc-zh#53] [Q&A]
> 本帖子长期更新微信群友关于 MLC 及 TVM 的部分讨论话题，希望帮助遇到类似问题的朋友 Get 一些启发 💡，有朝一日我们能沉淀出一个社区的 FAQ 文档！

1. **Q：跑 CPU ResNet18，TVM 比 PyTorch 慢了近一倍，合理吗？**
    A：应该是哪里出了点问题，ResNet50 上的结果应该是比 torch 快 3 倍左右
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h46friicjaj211u082my2.jpg)

2. **Q：请问 TVM 使用 cuda 跑一个结果后，如何保证运行完呢，类似于 PyTorch 里 torch.cuda.synchronize，我想测试耗时**
    A：TVM cuda 是非阻塞式的，你拿到 output 里面值的时候就运行完了。可以用 module.benchmark API 测量标准运行时间

3. **Q：如何跑通带 metal 的 iOS Demo？我这边带 bitcode 的 runtime 库还没有编译出来**
    A：可以参考 [Using Swift and Apache TVM to Develop ML Apps for the Apple Ecosystem](https://www.tvmcon.org/events/using-swift-and-apache-tvm-to-develop-ml-apps-for-the-apple-ecosystem/)

4. **Q：为什么会出现这个报错：**`ERROR: Could not install packages due toan OSError: HTTPSConnectionPool (host=' github.com', port=443)……`
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h46g5om60tj20zd09yjya.jpg)
    A：GitHub 被墙了

5. **Q：大家都是如何调试 TVM 的，像这种不会说到底第几行出错了**：
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h46g8am9cmj21h30lawgv.jpg)
A：目前在 notebook 中的报错信息还有些问题，在 Python 中运行是没有问题的

6. **Q：环境安装遇到这个报错：**
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h46gb1howvj20m803ldg8.jpg)
    A：python 升级到 3.7 及以上

7. **Q：出现报错 `DiagnosticError: Traceback (most recent call last)...`**
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h47f5s18srj21bn0u0jy6.jpg)
    A：函数里面有使用的`max`函数，应该用`tir.max`，且与 0 比较时要用`tir.float(0)`

8. **Q：schedule 中的 vectorize 和 parallel 有什么区别？**
    A：数据级并行和线程级并行

9. **Q：类似这样的函数，必须要指明 A B C 的 shape 吧？**
![](https://tva1.sinaimg.cn/large/e6c9d24ely1h4d8myp17oj20kv0ca40b.jpg)
    A：是的，必须要给定 shape

更多相关讨论，持续更新中 🙌🏻

**Comment by fairytale0828** [comment]
你好，微信群如何添加

**Comment by L220220** [comment]
请问问题6中我的python版本已经是3.8了依然报这个错怎么解决呀
